cdap-client/src/main/java/co/cask/cdap/client/ScheduleClient.java

```diff
@@ -70,6 +74,49 @@ public ScheduleClient(ClientConfig config) {
   }
 
   @Deprecated
+  public void add(Id.Schedule schedule, ScheduleInstanceConfiguration configuration) throws IOException,
+    UnauthenticatedException, NotFoundException, UnauthorizedException, AlreadyExistsException {
+    add(schedule.toEntityId(), configuration);
+  }
+
+  public void add(ScheduleId scheduleId, ScheduleInstanceConfiguration configuration) throws IOException,
+    UnauthenticatedException, NotFoundException, UnauthorizedException, AlreadyExistsException {
+
+    String path = String.format("apps/%s/versions/%s/schedules/%s",
+                                scheduleId.getApplication(), scheduleId.getVersion(), scheduleId.getSchedule());
```

Comment:
With your current command syntax, that is possible. What does scheduleId.getVersion() return if you don't specify a version? Will the endpoint work with that? Or would you have to call the non-versioned endpoint?